src/librbd/io/ImageRequestWQ.cc

```diff
@@ -311,6 +313,26 @@ void ImageRequestWQ<I>::aio_write(AioCompletion *c, uint64_t off, uint64_t len,
     trace.event("init");
   }
 
+  //block overlapping IOs
+  uint64_t tid = 0;
+  if (native_async && (m_image_ctx.non_blocking_aio || writes_blocked())) {
+    {
+      std::lock_guard locker{ordering_lock};
```

Comment:
Nit: why does this need a separate lock?

Reply:
The lock is for `m_queued_or_blocked_io_tids` and `m_blocked_ios` that are being used below. `m_last_tid` is now atomic